nomad/eval_endpoint.go

```diff
@@ -766,3 +767,35 @@ func (e *Eval) Allocations(args *structs.EvalSpecificRequest,
 		}}
 	return e.srv.blockingRPC(&opts)
 }
+
+// cancelCancelableEvals pulls a batch of cancelable evaluations from the eval
+// broker and updates their status to canceled.
+func cancelCancelableEvals(srv *Server) error {
```

Comment:
Note for reviewers: this is used by both the `Eval.Ack` endpoint and the leader's reaper, but we don't have a great spot for it in the code. Definitely open to suggestions for a better spot if you've got one.

Reply:
Short of a sizable refactor, I can't think of anywhere better.

Evaluations themselves have some methods for deriving _new_ evals, but that's not what's happening here.

The scheduler has a little `setStatus(...)` func that does effectively this ... and it's jammed in a `util.go` file, so clearly we don't have a good place to put eval munging. :sweat_smile:

Reply:
Moved into leader.go because `Eval.Ack` is no longer using it.